### ISSUE TITLE
Should Container.Collections be renamed to Container.Collection?

### ISSUE THREAD

**Issue by dotnetjunkie** [simpleinjector/SimpleInjector#556]
Simple Injector v4.1 introduced the `Container.Collections` property to group collection-based registrations in #517. From a readability standpoint, however, it might make more sense to rename this property to `Container.Collection`. Take the following example for instance:

``` c#
container.Collections.Register<ILogger>(typeof(ILogger).Assembly);

container.Collections.Append<ILogger, FileLogger>();
container.Collections.Append<ILogger, SqlLogger>();
```

These `Register` and `Append` methods however do not register or append _collections_, but actually a single _collection_. The following, therefore, does make more sense:

``` c#
container.Collection.Register<ILogger>(typeof(ILogger).Assembly);

container.Collection.Append<ILogger, FileLogger>();
container.Collection.Append<ILogger, SqlLogger>();
```

If the change is made, it should be made with the introduction of v4.3, because in the v4.2 and v4.1 releases, the `Collections` grouper only contains a few little used `Append` and `Create` methods, while with #542  in v4.3, `RegisterCollection` is moved to the grouper, which means a much larger group of developers will start to use the new syntax. Therefore, making the change _after_ v4.3 is of the table.

**Comment by dotnetjunkie** [comment]
To minimize the pain, I could even imagine to push this change to 4.1.3 and 4.2.3 as well.

**Comment by TheBigRic** [comment]
I totally agree. Naming this collection of methods `Collection` instead of `Collections` is far better and should be done now, because after this point the breaking change becomes way too big.
